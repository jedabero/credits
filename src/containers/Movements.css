.movementrow {
  display: grid;
  grid-template-columns: repeat(8, 1fr) 2fr;
}

.movementrow > input {
  display: block;
  min-width: 0;
}
