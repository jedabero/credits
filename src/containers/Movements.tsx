import { useCallback, useState } from 'react';
import { EntityId } from '@reduxjs/toolkit';

import MovementForm from '../components/MovementForm';
import MovementsTable from '../components/MovementsTable';

import './Movements.css';

type MovementProps = {
  cardId: EntityId
  onViewMovementHistory: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
};

function Movements({ cardId, onViewMovementHistory }: MovementProps) {
  const [movementIdBeingEdited, setMovementId] = useState<EntityId | undefined>();
  const clearMovementId = useCallback(() => setMovementId(undefined), [setMovementId]);
  return (
    <div className="movements">
      <h4>Movements</h4>
      <MovementForm
        cardId={cardId}
        movementId={movementIdBeingEdited}
        clearMovementId={clearMovementId}
      />
      <MovementsTable cardId={cardId} onEdit={setMovementId} onView={onViewMovementHistory} />
    </div>
  );
}

export default Movements;
