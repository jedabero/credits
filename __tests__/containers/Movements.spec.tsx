/* eslint-disable import/no-extraneous-dependencies */
import React, { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import Movements from '../../src/containers/Movements';

describe('Movements test', () => {
  const renderer = createRenderer();
  let output: ReactElement;
  test('render Movements', () => {
    renderer.render(<Movements cardId="CARD ID" onViewMovementHistory={() => {}} />);
    output = renderer.getRenderOutput<ReactElement>();
    expect(output).toMatchSnapshot();
  });
  test('set movementId', () => {
    output.props.children[2].props.onEdit('SOME ID');
    output = renderer.getRenderOutput<ReactElement>();
    expect(output.props.children[1].props.movementId).toBe('SOME ID');
    expect(output).toMatchSnapshot();
  });
  test('call clearMovementId', () => {
    output.props.children[1].props.clearMovementId();
    output = renderer.getRenderOutput<ReactElement>();
    expect(output.props.children[1].props.movementId).toBeNull();
    expect(output).toMatchSnapshot();
  });
});
