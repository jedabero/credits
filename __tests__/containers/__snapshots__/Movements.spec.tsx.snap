// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movements test call clearMovementId 1`] = `
<div
  className="movements"
>
  <h4>
    Movements
  </h4>
  <MovementForm
    cardId="CARD ID"
    clearMovementId={[Function]}
    movementId={null}
  />
  <MovementsTable
    cardId="CARD ID"
    onEdit={[Function]}
    onView={[Function]}
  />
</div>
`;

exports[`Movements test render Movements 1`] = `
<div
  className="movements"
>
  <h4>
    Movements
  </h4>
  <MovementForm
    cardId="CARD ID"
    clearMovementId={[Function]}
    movementId={null}
  />
  <MovementsTable
    cardId="CARD ID"
    onEdit={[Function]}
    onView={[Function]}
  />
</div>
`;

exports[`Movements test set movementId 1`] = `
<div
  className="movements"
>
  <h4>
    Movements
  </h4>
  <MovementForm
    cardId="CARD ID"
    clearMovementId={[Function]}
    movementId="SOME ID"
  />
  <MovementsTable
    cardId="CARD ID"
    onEdit={[Function]}
    onView={[Function]}
  />
</div>
`;
